Realm.Shared/linq/RealmResults.cs

```diff
@@ -295,16 +296,16 @@ internal static class RealmResultsNativeHelper
     {
         internal interface Interface
         {
-            void NotifyCallbacks(NativeResults.CollectionChangeSet? changes, NativeException? exception);
+            void NotifyCallbacks(ResultsHandle.CollectionChangeSet? changes, NativeException? exception);
         }
 
         #if __IOS__
-        [ObjCRuntime.MonoPInvokeCallback(typeof(NativeResults.NotificationCallback))]
+        [ObjCRuntime.MonoPInvokeCallback(typeof(ResultsHandle.NotificationCallback))]
         #endif
-        internal static void NotificationCallback(IntPtr managedResultsHandle, PtrTo<NativeResults.CollectionChangeSet> changes, PtrTo<NativeException> exception)
+        internal static void NotificationCallback(IntPtr managedResultsHandle, IntPtr changes, IntPtr exception)
         {
             var results = (Interface)GCHandle.FromIntPtr(managedResultsHandle).Target;
-            results.NotifyCallbacks(changes.Value, exception.Value);
+            results.NotifyCallbacks(new PtrTo<ResultsHandle.CollectionChangeSet>(changes).Value, new PtrTo<NativeException>(exception).Value);
```

Comment:
This `NotificationCallback ` bothers me a little bit I am wondering if it is robust if the callbacks triggered anything asynchronous because `new PtrTo<ResultsHandle.CollectionChangeSet` Will now go out of scope at the end of the notify callbacks call. I am not certain if this is any more dangerous than what we had before, however.